src/Components/CreateImageWizardV2/ValidatedTextInput.tsx

```diff
@@ -31,6 +31,10 @@ interface StateValidatedTextInputPropTypes extends TextInputProps {
   ariaLabel: string | undefined;
   helperText: string | undefined;
   validator: (value: string | undefined) => boolean;
+  asyncValidator?: (
+    value: string,
+    callback: (isValid: boolean) => void
+  ) => void;
```

Comment:
We now have second validator prop, that allows for async validation, when provided, it waits for the async validation to finish to mark validation success.